sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala

```diff
@@ -90,6 +91,46 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
           defaultMinimalPartitionCount
         )
     }
+
+    cosmosPartitioningConfig.feedRangeFiler match {
+      case Some(epkRangesInScope) => {
+        val overlappingPartitions = inputPartitions
+          .filter(inputPartition => {
+            epkRangesInScope.exists(epk => SparkBridgeImplementationInternal.doRangesOverlap(
+              epk,
+              inputPartition.feedRange))
+          })
+
+        val returnValue = new ArrayBuffer[CosmosInputPartition]()
+
+        overlappingPartitions.foreach(inputPartition => {
```

Comment:
Probably can combine with the above so only need do one time loop

Reply:
Possible - but not worth it. In the main scenario individual feedranges in the filter will be for a single logical partition - so never more than one physical partition epk-range would be overlapping. So, I intentionally optimized for simplicity here.